### DISCUSSION TITLE
top level await in pages script[setup] -> errors on quick navigation

### DISCUSSION THREAD

**Discussion by quarkus** [nuxt/nuxt#22727] [Questions] [answered]
While debugging i was trying to reduce complexity until i arrived at  a point where i am not sure what to expect anymore.

Simple Example:
https://stackblitz.com/edit/nuxt-starter-tkdxqq?file=pages%2Fa.vue

Two Pages A & B, both awaiting a Promise to resolve within their setup block (to simulate some blocking requests).

Both pages link to another and everything works fine...
**UNTIL you start binge clicking** (or just clicking faster than the timeout) which breaks the entire application.
So it seems that the component that is still waiting is removed from DOM before the Promise has resolved.. but thats just a guess.

I was going back and forth through the docs and figured that this should actually work since nuxt / the router uses Suspense internally.

https://nuxt.com/docs/getting-started/data-fetching#suspense

Am i right / wrong expecting two much ?

Thanks in advance.

**Comment by olivervorasai** [comment, marked as answer]
Related issues:
https://github.com/nuxt/nuxt/issues/13309
https://github.com/nuxt/nuxt/issues/13350
